snuba/datasets/dataset.py

```diff
@@ -40,3 +40,9 @@ def __init__(self, *, all_entities: Sequence[EntityKey]) -> None:
 
     def get_all_entities(self) -> Sequence[Entity]:
         return [get_entity(entity_key) for entity_key in self.__all_entities]
+
+    def __repr__(self) -> str:
+        # import down here bc of circular import
+        from snuba.datasets.factory import get_dataset_name
+
+        return f"get_dataset({repr(get_dataset_name(self))})"
```

Comment:
Why isn't this `repr(self)`? Why wrap it in `get_dataset`?

Reply:
repr(self) would return an object pointer since its not a dataclass.
I experimented with `f"Dataset(all_entities={repr(self.__all_entities)})"` because of what you said, but I think this would mess up the subclasses. Which is why I prefer the current change. A better solution could be to make it a dataclass so it has proper repr, but that seems like too much work.